recursion/recurse.rb

```diff
@@ -1,10 +1,22 @@
 def fact(n)
+  return 1 if n == 1 || n == 0 # Base case
+  n * fact(n - 1) # Recursive case
 end
 
 def fib(n)
+  return 1 if n == 1 || n == 2 # Base case
+  fib(n - 1) + fib(n - 2) # Recursive case
 end
 
 def pal(s)
+  return true if s.length == 1 || s.length == 0 # Base Case
+  if s[0] == s[-1]
+    s[0] = ''
+    s[-1] = ''
+    pal(s) # Recursive case
+  else
+    false
```

Comment:
Would be nice to test for this first

`return false if s[0] != s[-1]`

because typically we test all base cases first, then recursive cases.